apps/server/src/modules/user-migration/service/user-migration.service.spec.ts

```diff
@@ -0,0 +1,144 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { createMock, DeepMocked } from '@golevelup/ts-jest';
+import { MikroORM } from '@mikro-orm/core';
+import { setupEntities } from '@shared/testing';
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { EntityNotFoundError } from '@shared/common';
+import { SystemService } from '@src/modules/system/service';
+import { SystemDto } from '@src/modules/system/service/dto/system.dto';
+import { OauthConfigDto } from '@src/modules/system/service/dto/oauth-config.dto';
+import { UserMigrationService } from './user-migration.service';
+import { PageContentDto } from './dto/page-content.dto';
+import { PageTypes } from '../interface/page-types.enum';
+
+@Injectable()
+export class UserMigrationServiceSpy extends UserMigrationService {
+	public readonly PROCESS_MIGRATION_BASE_URL: string = '/api/v3/oauth/migration';
+
+	public getOauthLoginUrlSpy(system: SystemDto, postLoginUri?: string): string {
+		return this.getOauthLoginUrl(system, postLoginUri);
+	}
+}
+
+describe('MigrationService', () => {
+	let module: TestingModule;
+	let service: UserMigrationServiceSpy;
+	let systemService: DeepMocked<SystemService>;
+	let orm: MikroORM;
+
+	beforeAll(async () => {
+		module = await Test.createTestingModule({
+			providers: [
+				{
+					provide: SystemService,
+					useValue: createMock<SystemService>(),
+				},
+				UserMigrationServiceSpy,
+			],
+		}).compile();
+		systemService = module.get(SystemService);
+		service = module.get(UserMigrationServiceSpy);
+		orm = await setupEntities();
+	});
+
+	afterAll(async () => {
+		await module.close();
+		await orm.close();
+	});
+
+	const setup = () => {
+		const defaultOauthConfigDto: OauthConfigDto = new OauthConfigDto({
+			clientId: '12345',
+			clientSecret: 'mocksecret',
+			tokenEndpoint: 'http://mock.de/mock/auth/public/mockToken',
+			grantType: 'authorization_code',
+			scope: 'openid uuid',
+			responseType: 'code',
+			authEndpoint: 'http://mock.de/mock/auth',
+			provider: 'mock_provider',
+			logoutEndpoint: 'mock_logoutEndpoint',
+			issuer: 'mock_issuer',
+			jwksEndpoint: 'mock_jwksEndpoint',
+			redirectUri: 'http://mock.de/mock/redirect',
+		});
+		const mockSystem: SystemDto = new SystemDto({
+			type: 'oauth',
+			url: 'http://mockhost:3030/api/v3/oauth',
+			alias: 'Iserv',
+			oauthConfig: defaultOauthConfigDto,
+		});
+
+		const proceedUrl: string = service.getOauthLoginUrlSpy(mockSystem);
```

Comment:
this makes no fucking sense

Reply:
it does. it creates the properly built redirectURL. as said before, not an indeal solution but but it ensures proper checks

Reply:
...fine